proposals/kapp/001-crd-upgrade-safety/README.md

```diff
@@ -0,0 +1,95 @@
+---
+title: "Adding CRD Upgrade Safety to kapp"
+authors: [ "Bryce Palmer <bpalmer@redhat.com>" ]
+status: "in review"
+approvers: [ "Carvel Maintainers" ]
+---
+
+# Adding CRD Upgrade Safety to kapp
+
+## Problem Statement
+CustomResourceDefinitions (CRDs) are an integral part of extending the Kubernetes API and are often applied to a cluster alongside a controller that manages them.
+
+Since CRDs are global and only one instance can exist on the cluster at a time, it is important to ensure that updates to a CRD are performed safely.
+This means preventing data loss and breaking of clients/workloads that may be using the API.
+Failure to safely update a CRD may result in:
+- CustomResources (CRs) that currently exist in the object store no longer being valid, causing extensions to fail reconciliation of the existing CRs
```

Comment:
nit: my brain is bending around "no longer being valid".  It's probably a passive-voice thing, but I get that the focus is that the failure condition has been triggered but won't become evident until later (when it's too late).
Maybe
```suggestion
- invalidation of existing CustomResources (CRs) in the object store, causing extensions to fail reconciliation of the existing CRs
```
?

Reply:
Updated in https://github.com/carvel-dev/carvel/pull/729/commits/af7e1f16775c19143b093e4e8cda347fe94edb02 - does this sound better?